day_7/10_speckled_frogs.rb

```diff
@@ -0,0 +1,51 @@
+require 'to_words'
+
+def speckled_frogs(total_frogs=10, include_words=false)
+  total_frogs.times do
+    case include_words
+    when false # print numbers instead of words
+      if total_frogs > 2
+        puts "#{total_frogs} speckled frogs sat on a log eating some most delicious bugs."
+        puts "One jumped in the pool where its nice and cool,"
+        total_frogs -= 1
+        puts "then there were #{total_frogs} speckled frogs."
+      elsif total_frogs == 2
+        puts "#{total_frogs} speckled frogs sat on a log eating some most delicious bugs."
+        puts "One jumped in the pool where its nice and cool,"
+        total_frogs -= 1
+        puts "then there was #{total_frogs} speckled frog."
+      else
+        puts "#{total_frogs} speckled frog sat on a log eating some most delicious bugs."
+        puts "One jumped in the pool where its nice and cool,"
+        total_frogs -= 1
+        puts "then there were no speckled frogs!"
+      end
+    when true #print words instead of numbers
+      total_frogs.times do
+        if total_frogs > 2
+          puts "#{total_frogs.to_words.capitalize} speckled frogs sat on a log eating some most delicious bugs."
+          puts "One jumped in the pool where its nice and cool,"
+          total_frogs -= 1
+          puts "then there were #{total_frogs.to_words} speckled frogs."
+        elsif total_frogs == 2
+          puts "#{total_frogs.to_words.capitalize} speckled frogs sat on a log eating some most delicious bugs."
+          puts "One jumped in the pool where its nice and cool,"
+          total_frogs -= 1
+          puts "then there was #{total_frogs.to_words} speckled frog."
+        else
+          puts "#{total_frogs.to_words.capitalize} speckled frog sat on a log eating some most delicious bugs."
+          puts "One jumped in the pool where its nice and cool,"
+          total_frogs -= 1
+          puts "then there were no speckled frogs!"
+        end
+      end
+    end
+  end
+end
+
```

Comment:
Very nice job including multiple extensions! Something that may help with the 'bulk' of this code (here, I mean lots of string typing) could be to break out a separate function that just `puts` your repeated lines. Anytime you are seeing repeated text, or logic, that can be a cue to break it out into a method for re-useability.